### ISSUE TITLE
How to access GCS bucket from another project?

### ISSUE THREAD

**Issue by the1plummie** [GoogleCloudPlatform/gcloud-java#1073]
So I have files in a bucket `BK1` created in gcloud project `A`, and I have appengine service running in project `B`. How do I configure `BK1` bucket (or object) permissions so B's appengine service can access those files?

My sample code looks like this
```
        Storage storage = StorageOptions.defaultInstance().service();
        String filename = "foo.json";
        Blob blob = storage.get(BlobId.of("BK1", filename));
        if (blob == null) {
            return null;
        }
        System.out.println("blob content: " + new String(blob.content()));
```
I'm getting "login required" error. 

I tried adding app engine default service account to BK1's permission, still not working either.
Any pointer would be appreciated.
Thx!

**Comment by mziccard** [comment]
Hi @the1plummie,

which version of `gcloud-java` are you using? We have seen this "login required" issue before (see [here](https://github.com/GoogleCloudPlatform/gcloud-java/issues/1007)). This should be fixed in 0.2.3 now.

Also remember that to get blob content/metadata the app engine service account must also have (at least) READER permission to the blob itself.

**Comment by the1plummie** [comment]
Good point. I was using version 0.2.0. Confirmed it's fixed in 0.2.3. Thanks! 👍
